app/src/main/java/com/cp/tasky/core/navigation/NavGraph.kt

```diff
@@ -0,0 +1,51 @@
+package com.cp.tasky.core.navigation
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import com.cp.tasky.auth.login.presentation.LoginScreenRoot
+
+@Composable
+fun SetUpNavGraph(
+    modifier: Modifier = Modifier,
+    navHostController: NavHostController = rememberNavController(),
+    startDestination: String = Screen.Login.route // Replace with SplashScreen as start destination later
+) {
+
+    NavHost(
+        modifier = modifier,
+        navController = navHostController,
+        startDestination = startDestination
+    ) {
+        composable(route = Screen.Splash.route) {
+            // TODO: Add Splash Screen
+        }
+
+        composable(route = Screen.Login.route) {
+            LoginScreenRoot()
+        }
+
+        composable(route = Screen.Register.route) {
+            // TODO: Add Register Screen
+        }
```

Comment:
I like to work with one navigation graph per feature, since then you can conveniently pop the whole graph from the back stack when logging in for example without risking there are some screens left on the back stack.

Works just like defining routes:
```
@Serializable
data object AuthGraph

NavHost(...) {
    navigation<AuthGraph>(startDestination = LoginRoute) {
        composable<LoginRoute> {
        
        }
        ...
    }
}
```